flink-table-store-core/src/main/java/org/apache/flink/table/store/table/AbstractFileStoreTable.java

```diff
@@ -179,4 +188,25 @@ public TableCommitImpl newCommit(String commitUser) {
                 options().writeOnly() ? null : store().newExpire(),
                 options().writeOnly() ? null : store().newPartitionExpire(commitUser));
     }
+
+    private void tryTimeTravel(Options options) {
+        if (options.contains(CoreOptions.TIME_TRAVEL_SNAPSHOT_ID)) {
```

Comment:
In dynamicOptions, but not equals in old options, do time travel?

Reply:
If we keep using `TIME_TRAVEL_SNAPSHOT_ID`, it's not a problem because when a table is created, this option won't appear, and if a time travel call pass this option, here it'll be cleaned.

If we use `scan.from-snapshot` and `scan.from-timstamp` as time travel mark, I think we should always use new value (new snapshotId or timestamp) as target. Furthermore, I think when the `TableScan` scan to next snapshot, the `TableSchema` should also be replaced to that snapshot's schema. 

What do you think?

Reply:
I may should know why we need to introduce `TIME_TRAVEL_SNAPSHOT_ID`.